qiskit_experiments/calibration_management/save_utils.py

```diff
@@ -0,0 +1,253 @@
+    control_channel_map: ControlChannelMap
+    schedules: List[ScheduleModelV1] = field(default_factory=list)
+    parameters: List[ParameterModelV1] = field(default_factory=list)
+    schema_version: str = "1.0"
+
+    @classmethod
+    def apply_schema(cls, data: Dict[str, Any]):
+        """Consume dictionary and returns canonical data model."""
+        in_data = {}
+        for key, value in data.items():
+            if key == "schedules":
+                value = list(map(ScheduleModelV1.apply_schema, value))
+            if key == "parameters":
+                value = list(map(ParameterModelV1.apply_schema, value))
+            in_data[key] = value
+        return CalibrationModelV1(**in_data)
+
+
+def to_dict(
+    cals: Calibrations,
+    most_recent_only: bool = True,
+) -> Dict[str, Any]:
+    """A helper function to convert calibration data into dictionary.
+
+    Args:
+        cals: A calibration instance to save.
+        most_recent_only: Set True to save calibration parameters with most recent time stamps.
+
+    Returns:
+        Canonicalized calibration data in dictionary format.
+    """
+    # This can dispatch canonicalize function as version evolves.
+    model = _canonicalize_calibration_data_v1(cals, most_recent_only)
+    return asdict(model)
+
+
+def from_dict(
+    cal_data: Dict[str, Any],
+) -> Calibrations:
+    """A helper function to build calibration instance from canonical dictionary.
+
+    Args:
+        cal_data: Calibration data that conforms to the predefined data schema.
+
+    Returns:
+        Calibration instance.
+
+    Raises:
+        ValueError: When input data model version is no longer supported.
+        KeyError: When input data model doesn't conform to data schema.
+    """
+    # Apply schema for data field validation
+    try:
+        version = cal_data["schema_version"]
+        if version == "1.0":
+            model = CalibrationModelV1.apply_schema(cal_data)
+        else:
+            raise ValueError(
+                f"Loading calibration data with schema version {version} is no longer supported. "
+                "Use the same version of Qiskit Experiments at the time of saving."
+            )
+    except (KeyError, TypeError) as ex:
+        raise KeyError(
+            "Loaded data doesn't match with the defined data schema. "
+            "Check if this object is dumped from the Calibrations instance."
+        ) from ex
+
+    # This can dispatch loading mechanism depending on schema version
+    cals = Calibrations(
+        coupling_map=model.device_coupling_graph,
+        control_channel_map=model.control_channel_map.chan_map,
+        backend_name=model.backend_name,
+        backend_version=model.backend_version,
+    )
+
+    # Add schedules
+    for sched in model.schedules:
+        cals.add_schedule(
+            schedule=sched.data,
+            qubits=sched.qubits if len(sched.qubits) != 0 else None,
+            num_qubits=sched.num_qubits,
+        )
```

Comment:
Does this keep any parameter links that we initially had?

Reply:
Yes. QPY preserves parameter UUID and it serializes the schedule through QPY format.

Reply:
Ok good. Thanks, do we have a test on this?